Documentation/ceph-object-bucket-notifications.md

```diff
@@ -0,0 +1,169 @@
+A CBN define: topic, events and filters of a bucket notifications. Meaning, it defines what triggers the notification, and where to send the notifications to. However, to complete the definition of the bucket notification, it must be associated with an OBC. This is done outside of the CBN resource, via adding labels to an OBC with the following format:
+```yaml
+bucket-notification-<notification name>: <notification name>
+```
+> Note that the CBT, CBN and OBC must all belong to the same namespace
+
+## Notification Reliability and Delivery
+Notifications may be sent synchronously, as part of the operation that triggered them. In this mode, the operation is acked only after the notification is sent to the topic’s configured endpoint, which means that the round trip time of the notification is added to the latency of the operation itself.
+> Note that the original triggering operation will still be considered as successful even if the notification fail with an error, cannot be delivered or times out
+
+Notifications may also be sent asynchronously. They will be committed into persistent storage and then asynchronously sent to the topic’s configured endpoint. In this case, the only latency added to the original operation is of committing the notification to persistent storage.
+> Note that if the notification fail with an error, cannot be delivered or times out, it will be retried until successfully acked
+
+## Sample
+
+### CBT Custom Resource
+```yaml
+apiVersion: objectbucket.io/v1alpha1
+kind: CephBucketTopic
+metadata:
+  name: my-topic [1]
+  namespace: my-app-space [2]
+spec:
+   # use one of the following endpoints
+  endpoint: http://my-notification-endpoint:8080 [3]
+  # endpoint: http://my-notification-endpoint:8080/my-topic
+  # endpoint: https://my-notification-endpoint:8443
+  # endpoint: amqp://my-rabbitmq-service:5672/vhost1
+  # endpoint: amqps://my-rabbitmq-service:5671/vhost1
+  # endpoint: kafka://my-kafka-service:9092
+  objectStoreName: my-store [4]
+  objectStoreNamespace: rook-ceph [5]
+  opaqueData: my@email.com [6]
+  persistent: false [7]
+  http: [8]
+    verifySSL: false [9]
+  # amqp: [10]
+  #   verifySSL: false [11]
+  #   ackLevel: broker [12]
+  #   exchange: my-exchange [13]
+  # kafka: [14]
+  #   verifySSL: false [15]
+  #   ackLevel: broker [16]
+  #   useSSL: false [17]
+
+```
+1. `name` of the `CephBucketTopic`
+   - in case of AMQP endpoint, the name is used for the AMQP topic (“routing key” for a topic exchange)
+   - in case of Kafka endpoint, the name is used as the Kafka topic
+1. `namespace`(optional) of the `CephBucketTopic`. Should match the namespace of the CBN associated with this CBT, and the OBC with the label referencing the CBN
+1. `endpoint` to which to send the notifications to. the identity of the endpoint in formatted as a URI with the following schema options: `http[s]`, `amqp[s]`, `kafka`. The schema of the endpoint must match the type of the endpoint spec provided (see below)
```

Comment:
To be consistent with other rook docs, how about separating the settings with a `:` and capitalizing?
```suggestion
1. `endpoint`: Endpoint to which to send the notifications to. The identity of the endpoint in formatted as a URI with the following schema options: `http[s]`, `amqp[s]`, `kafka`. The schema of the endpoint must match the type of the endpoint spec provided (see below)
```

Reply:
* will capitilize
* i tried to follow the OBC example: https://github.com/rook/rook/blob/master/Documentation/ceph-object-bucket-claim.md#obc-custom-resource :
```markdown
1. `name` of the `ObjectBucketClaim`. This name becomes the name of the Secret and ConfigMap.
1. `namespace`(optional) of the `ObjectBucketClaim`, which is also the namespace of the ConfigMap and Secret.
1. `bucketName` name of the `bucket`.
...
```

Reply:
@travisn - will do the capitalization

regarding the other comment, I followed the same style used in the parameter description of the OBC:
https://github.com/rook/rook/blob/master/Documentation/ceph-object-bucket-claim.md#obc-custom-resource

Reply:
Not a big deal, following the pattern in the OBC doc is fine.